Lib/importlib/_bootstrap.py

```diff
@@ -1005,18 +1007,15 @@ def _find_and_load_unlocked(name, import_):
     if spec is None:
         raise ModuleNotFoundError(_ERR_MSG.format(name), name=name)
     else:
-        if parent and parent_spec:
-            # add child to the parent _uninitialized_submodules so that we can
-            # flag it in case it has a circular import on the parent
-            _uninitialized = getattr(parent_module.__spec__,
-                                    '_uninitialized_submodules',
-                                    [])
-            parent_spec._uninitialized_submodules = _uninitialized + [child]
+        if parent_spec:
+            # Temporarily add currently imported child to parent's
```

Comment:
Hum, shouldn't it more correct to say "child we are going to import" instead of "currently imported child"? As the child isn't actually imported, we are going to try to import it.

Reply:
The error occurs when the child (let's say `a.b.c`) tries to import the parent (`a.b`), but the parent also imports the child, which results in the child not importing the parent, just the parent's parent without (`a`) and not setting the parent as an attribute (`a.b` will not be set). We want to be able to flag that error when we import the child (`a.b.c`) and provide an nicer message if it tries to access the parent (`a.b`, which is not set). Very confusing, I know :sweat_smile: